infrastructure/main.tf

```diff
@@ -36,37 +36,83 @@ locals {
   }
 }
 
+data "google_compute_network" "existing-network" {
+  count   = var.create_network ? 0 : 1
+  name    = var.network_name
+  project = var.project_id
+}
+
 module "custom-network" {
   source       = "terraform-google-modules/network/google"
   version      = "8.0.0"
   count        = var.create_network ? 1 : 0
   project_id   = var.project_id
   network_name = var.network_name
 
-  subnets = [
-    {
-      subnet_name           = var.subnetwork_name
-      subnet_ip             = var.subnetwork_cidr
-      subnet_region         = var.subnetwork_region
-      subnet_private_access = var.subnetwork_private_access
-      description           = var.subnetwork_description
-    }
-  ]
-
+  subnets = [for subnet_region, subnet_ip in zipmap(var.subnetwork_regions, local.subnetwork_cidr_ips) : {
```

Comment:
Actually, we can remove subnets all together and just enable auto-mode that will reduce the inputs.

Reply:
Auto created subnets do not have "Private Google Access" enabled when created. When I tested this out using auto-mode network, pods that were created in the (Autopilot) GKE cluster were unable to access the internet, and "Private Google Access" was not enabled on any of the auto-created subnets.

According to the documentation, the hack is to enable Cloud NAT, which auto enalbes PGA: https://cloud.google.com/nat/docs/nat-product-interactions#interaction-pga. It seems like this should get enabled (within the cluster's region here: https://github.com/GoogleCloudPlatform/ai-on-gke/blob/main/infrastructure/main.tf#L201), but as mentioned above I ran into that error testing. IMO I prefer the explicit PGA subnet creation to relying on the Cloud NAT side effect, as it's not immediately obvious that Cloud NAT has this effect.

Reply:
yeah i forget to delete the comment

Reply:
Sorry my other left in draft.

Let be it single region and single subnet only. This change will impact the existing scenarios which are already tested and validated. 

We can do this change after next release.. as we have code freeze by next week Tuesaday

@pwschuurman

Reply:
Sounds good. I'll port that complexity into a new PR.

I was thinking that as well, this PR started to get a bit too large.